src/betanc.jl

```diff
@@ -0,0 +1,164 @@
+const errmax = 1e-15
+
+#Compute tail of noncentral Beta distribution
+#Russell Lenth, Algorithm AS 226: Computing Noncentral Beta Probabilities,
+#Applied Statistics,Volume 36, Number 2, 1987, pages 241-244
+
+"""
+	ncbeta_tail(x,a,b,lambda)
+
+Compute tail of the noncentral beta distribution.
+Uses the recursive relation 
+```math
+I_{x}(a,b+1;0) = I_{x}(a,b;0) - \\Gamma(a+b)/\\Gamma(a+1)\\Gamma(b)x^{a}(1-x)^{b}
+```
+and ``\\Gamma(a+1) = a\\Gamma(a)`` given in Equation (26.5.16) in Abromowitz and Stegun(1965).
+"""
+function ncbeta_tail(x::Float64, a::Float64, b::Float64, lambda::Float64)
+    if x <= 0.0
+        return 0.0
+    elseif x >= 1.0
+        return 1.0
+    end
+    
+    c = 0.5*lambda
+    #Init series
+
+    beta = logabsbeta(a,b)[1]
+    temp = beta_inc(a,b,x)[1]
+    gx = (SpecialFunctions.beta_integrand(a,b,x,1.0-x))/a
+    q = exp(-c)
+    xj = 0.0
+    ax = q*temp
+    sumq = 1.0 - q
+    ans = ax
+
+    while true
+        xj += 1.0
+        temp -= gx
+        gx *= x*(a+b+xj-1.0)/(a+xj)
+        q *= c/xj
+        sumq -= q
+        ax = temp*q
+        ans += ax
+
+        #Check convergence
+        errbd = abs((temp-gx)*sumq)
+        if xj > 1000 || errbd < 1e-10
+            break
+        end
+    end
+    return ans

```

Comment:
can we compute both lower and upper tails?

Reply:
Is that needed? Like the paper hasn't written anything on computing upper and lower tails.

Reply:
Rather this approach uses a 2-way iteration from middle to top / middle to bottom to improve upon the error bound as mentioned in Documentation.

Reply:
Like the incomplete beta, this is used for computing the CDF of a probability distribution. If we're computing one tail, we might as well compute the other.

Reply:
Yes but in what way? 
This paper has only mentioned for `lambda < 54`.

Reply:
okay, lets leave it for the time being.